### PR TITLE
Add Ondatra lint to pr workflow

### PR DESCRIPTION
**Pull request by DanG100**

### CODE REVIEW

File: .github/workflows/pull_request.yml [hunk 1]
```diff
@@ -164,3 +164,9 @@ jobs:
       run: go install honnef.co/go/tools/cmd/staticcheck@latest
     - name: Run staticcheck
       run: GOGC=30 staticcheck ./...
+    - name: Get Ondatra linter
+      run: go install github.com/openconfig/ondatra/gnmi/migrator@latest
+    - name: Run Ondatra linter
+      run: |
+        migrator -json ./feature/... > lint.json
```
Comment (liulk) [inline .github/workflows/pull_request.yml:171]: Can we change the lint2annotation to read from stdin? Then we can just:

migrator -json ./feature/... | go run ./tools/lint2annotation

**Comment by liulk** [comment]
I like the look of this. @bstoll what do you think?

**Comment by marcushines** [comment]
This looks pretty cool nice daniel

**Review by greg-dennis — COMMENTED** [review]
Looks great

File: tools/lint2annotation/lint2annotation.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"strings"
+)
+
+type diag struct {
+	Category string `json:"category,omitempty"`
```
[the remark this answers is not in the document]
Comment (DanG100) [reply tools/lint2annotation/lint2annotation.go]: removed

**Comment by bstoll** [comment]
This is cool.  Should this use `::warning` or `::notice` for now?  I think calling these errors is a lot to drop on us all at once.

**Comment by DanG100** [comment]
> This is cool. Should this use `::warning` or `::notice` (my preference) for now? I think calling these errors is a lot to drop on us all at once.

done, and made a flag to make it easier to change

**Review by bstoll — APPROVED** [review]

**Pull request merged**